.github/workflows/os-compatibility-test.yaml

```diff
@@ -2,22 +2,35 @@ name: OS Compatibility Test
 run-name: OS Compatibility Test
 on:
   pull_request:
-    paths-ignore:
-      - 'docs/**'
-      - 'web/**'
-      - 'rfd/**'
-      - '**/*.md*'
   merge_group:
-    paths-ignore:
-      - 'docs/**'
-      - 'web/**'
-      - 'rfd/**'
-      - '**/*.md*'
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      this-workflow: ${{ steps.changes.outputs.this-workflow }}
+      binary-sources: ${{ steps.changes.outputs.binary-sources }}
+    steps:
+    - uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
+      id: changes
+      with:
+        filters: |
+          this-workflow:
+            - '.github/workflows/os-compatibility-test.yaml'
+          binary-sources:
```

Comment:
Is this sufficient to capture the requirements? What about Rust files, shell scripts, and Docker files that are used in the build? Or e ref updates? Our build process is such a mess that I don't think that we can whitelist file extensions and still have the same level of coverage that trigger currently has.